### ISSUE TITLE
Nothing happens when joining

### ISSUE THREAD

**Issue by dmdque** [davidstump/SwiftPhoenixClient#61]
I'm using a simple Phoenix instance, and have a simple channel join event working in js. I'm using the following on iOS:

        let socket = Socket(domainAndPort: "localhost:4000", path: "socket", transport: "websocket")
        socket.join(topic: "room:lobby", message: Message(subject: "status", body: "joining")) { channel in
            let channel = channel as! Channel
            print("join")
        }

But nothing is logged on the server. I'm watching the traffic going through the server's port 4000 and am seeing some packets flow every time I trigger the above code.

Not sure if I should post this in the Phoenix Framework project as well.

**Comment by davidstump** [comment]
Well, that sounds less than ideal. I am out of town, but let me check this out when I get back before posting to Phoenix so we can confirm its not a problem on our end (which is more likely).

Thanks for the report!

**Comment by dmdque** [comment]
Just going to add some more information. When running the `Example/`, my logs get flooded with this:

```2016-11-28 16:11:28.449741 ChatExample[13051:3587693] [] __nw_connection_get_connected_socket_block_invoke 240 Connection has no connected handler
Error: Error Domain=NSPOSIXErrorDomain Code=61 "Connection refused" UserInfo={_kCFStreamErrorCodeKey=61, _kCFStreamErrorDomainKey=1}
socket closed: Optional("The operation couldn’t be completed. Connection refused")```

I'm using https://github.com/scouten/phoenix_chat_example/tree/update-deps instead of the original phoenix_chat_example since I was getting some build errors.

**Comment by davidstump** [comment]
Thanks for the added info - can you paste what you are seeing if anything on the logs for the Phoenix side?

**Comment by dmdque** [comment]
The server isn't logging anything.

`sudo tcpdump port 4000` also outputs nothing when using the example project.

**Comment by dmdque** [comment]
I tried the channel join with [Birdsong](https://github.com/sjrmanning/Birdsong), and the results are the same.

I've created an issue in the phoenix project. There's some additional info in there.

**Comment by davidstump** [comment]
Awesome. Sorry for the delay but thanks for checking with Birdsong as well.

**Comment by dmdque** [comment]
I managed to get Birdsong to work. I was missing line 3:

    let socket = Birdsong.Socket(url: URL(string: "http://10.88.111.2:4000/socket/websocket")!)
    socket.onConnect = {
        let channel = socket.channel("room:lobby", payload: ["user": "test"])
    }
    socket.connect()

It looks like such an innocent line, but apparently it's crucial.

Perhaps I'm missing something in my SwiftPhoenixClient code? Since they had nearly identical behaviour before, it seems I'm missing whatever line 3 is doing.

**Comment by davidstump** [comment]
In the initial code you provided from SwiftPhoenixClient the topic was `room:lobby` but the updated chat example uses `rooms:lobby`. Can you confirm for me that the topic in your Swift client matches whats in the channel?

**Comment by dmdque** [comment]
Yeah, I modified the chat example to use `room`. Also Birdsong is using `room` and works.

**Comment by davidstump** [comment]
I ran across a number of different issues opened with similar errors since iOS 10. One issue: https://github.com/facebook/react-native/issues/10027 had a solution that worked for me.  

product -> clean
and (hold down option key)
product -> 'clean build folder'

Prior to that I saw the same no connected handler error you referenced. After I could connect just as it should.

Let me know if this helps.

**Comment by dmdque** [comment]
I followed those steps but the behaviour is the same.

Can you confirm that the code in my first comment *should* work (and isn't missing anything)?

    let socket = SwiftPhoenixClient.Socket(domainAndPort: "10.88.111.2:4000", path: "socket", transport: "websocket")
    socket.join(topic: "room:lobby", message: SwiftPhoenixClient.Message(subject: "status", body: "joining")) { channel in
        let channel = channel as! SwiftPhoenixClient.Channel
        channel.on(event: "join") { message in
            print("You joined the room.\n")
        }
        print("join")
    }

**Comment by davidstump** [comment]
What I have in my sample running now is:

```
let socket = Socket(domainAndPort: "localhost:4000", path: "socket", transport: "websocket", params: ["user": "iDavid"])
var topic: String? = "rooms:lobby"

socket.join(topic: topic!, message: Message(subject: "status", body: "joining")) { channel in
      let chan = channel as! Channel
      
      chan.on(event: "phx_join") { message in
        self.chatWindow.text = "You joined the room.\n"
      }

...
```

Notably, I do not have SwiftPhoenixClient in front of my class names. Also, the example code is using phx_join to wait for the join event to succeed. However, the error messages should/would be different for either of those issues I believe.

**Comment by dmdque** [comment]
Seems functionally identical to me... hmm.

As for prefixing, that's due to the namespace collisions between SwiftPhoenixClient and Birdsong. I'm testing both in the same file.

I'm not sure what to try at this point.

**Comment by davidstump** [comment]
Based on a few of the other reports - it sounds like a bug in some users Xcode env since iOS 10. Happy to keep helping if I can, however, is there a reason SwiftPhoenixClient better suits your needs than Birdsong? Not trying to punt, just trying to get you back on track building something versus spending forever debugging :)

**Comment by dmdque** [comment]
I see, that's too bad. I've started implementing with Birdsong. I'll end my debugging of this here, for now. Thanks for your help!

**Comment by davidstump** [comment]
Not a problem! Best of luck with your project.
